src/components/explore/CategoricalMenu.svelte

```diff
@@ -0,0 +1,160 @@
+<script>
+  import { OptionMenu, Option, OptionDivider } from '@graph-paper/optionmenu';
+
+  import { store } from '../../state/store';
+
+  import KeySelectionControl from '../controls/KeySelectionControl.svelte';
+  import ColorSwatch from '../controls/ColorSwatch.svelte';
+
+  import { numericStringsSort } from '../../utils/sort';
+  import { numHighlightedBuckets } from '../../config/shared';
+
+  export let data;
+  export let activeBuckets;
+  export let bucketColorMap;
+  export let bucketOptions;
+
+  export let bucketSortOrder = (a, b) => (a < b ? 1 : -1);
+
+  function makeSelection() {
+    return function onSelection(event) {
+      store.setField('activeBuckets', event.detail.selection);
+    };
+  }
+
+  let showOptionMenu = false;
+  let coloredBuckets = [];
+  let everActiveBuckets = [];
+  let sortedImportantBuckets = [];
+  let sortedUnimportantBuckets = [];
+
+  if (bucketOptions.length > numHighlightedBuckets) {
+    showOptionMenu = true;
+    const lastDataset = data[data.length - 1];
+
+    coloredBuckets = Object.entries(lastDataset.counts)
+      .sort(([bucketAName, bucketAValue], [bucketBName, bucketBValue]) => {
+        const bucketValueDifference = bucketAValue - bucketBValue;
+        if (bucketValueDifference === 0) {
+          return bucketBName - bucketAName;
+        }
+        return bucketValueDifference;
+      })
+      .slice(-numHighlightedBuckets)
+      .map(([bucket]) => bucket);
+  }
+
+  let selectAllCategories = false;
+
+  $: if (showOptionMenu) {
+    everActiveBuckets = [...new Set([...activeBuckets, ...everActiveBuckets])];
+
+    // An important bucket is any bucket that:
+    //
+    //   (a) is colored
+    //   (b) is currently active
+    //   (c) has been active at some point in the past
+    //
+    // Rule (c) improves usability: if the user goes out of their way to enable a
+    // bucket that we consider to be unimportant, we should consider it to be
+    // important for the rest of the interaction. If we did not do this, the bucket
+    // would switch between the important group and the unimportant group as it's
+    // toggled, which would be annoying.
+    sortedImportantBuckets = [
+      ...new Set([...everActiveBuckets, ...coloredBuckets]),
+    ].sort(numericStringsSort);
+
+    // An unimportant bucket is any other bucket
+    sortedUnimportantBuckets = bucketOptions
+      .filter((bucket) => !sortedImportantBuckets.includes(bucket))
+      .sort(numericStringsSort);
+  }
+
+  function handleSelectAllCategories() {
+    store.setField('activeBuckets', bucketOptions);
+  }
+
+  $: selectAllCategories =
+    $store.activeBuckets.length === bucketOptions.length && true;
```

Comment:
Why is `&& true` necessary here?

Reply:
oops this is not right! thanks for catching this!